### PR TITLE
In the mobile menu, preview entry list for each category

### PR DESCRIPTION
**Pull request by MarcelGerber**
https://www.notion.so/Fill-up-mobile-menu-empty-space-with-entry-list-1f816b7560864ffa85b70bc2459a6154

## Before:
ourworldindata org_(iPhone X)

## After (iPhone X, iPad):
localhost_3099_(iPhone X)
localhost_3099_(iPad) (1)

These item previews will only be shown in the mobile menu, and only for categories that are not expanded. They are cutoff with "...".

### CODE REVIEW

File: site/client/SiteHeaderMenus.tsx [hunk 1]
```diff
@@ -306,6 +306,16 @@ const renderEntry = (entry: EntryMeta): JSX.Element => {
...
 
+const allEntries = (category: CategoryWithEntries): EntryMeta[] => {
+    // combine "direct" entries and those from subcategories
+    return [
+        ...category.entries,
+        ...flatten(
+            category.subcategories.map(subcategory => subcategory.entries)
+        )
+    ]
+}
```
Comment (danielgavrilov) [inline site/client/SiteHeaderMenus.tsx:317]: Nice refactoring!

**Comment by danielgavrilov** [comment]
@MarcelGerber It's great! 

How did you test this?! Did you mock the header menu JSON file?

**Comment by MarcelGerber** [comment]
> How did you test this?! Did you mock the header menu JSON file?

No, just running `yarn dev` and you get the whole site, including the header. Simple as that :)

**Comment by danielgavrilov** [comment]
Don't you need the Wordpress database to generate the header menu? https://github.com/owid/owid-grapher/blob/1e1a602e0d0989ba3cf414cad66e7bac400a6b70/site/server/siteBaking.tsx#L139-L142

**Comment by MarcelGerber** [comment]
Looks like `WORDPRESS_URL` defaults to owid.cloud, so the graphQL query is going there and everything just works

**Comment by danielgavrilov** [comment]
I had no idea we have a public Wordpress API. 😳

**Pull request merged**